lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -282,6 +282,47 @@ SPIRVToLLVMDbgTran::transTypeArrayNonSemantic(const SPIRVExtInst *DebugInst) {
   return Builder.createArrayType(Size, 0 /*align*/, BaseTy, SubscriptArray);
 }
 
+DICompositeType *
+SPIRVToLLVMDbgTran::transTypeArrayDynamic(const SPIRVExtInst *DebugInst) {
+  using namespace SPIRVDebug::Operand::TypeArrayDynamic;
+  const SPIRVWordVec &Ops = DebugInst->getArguments();
+  assert(Ops.size() >= MinOperandCount && "Invalid number of operands");
+  DIType *BaseTy =
+      transDebugInst<DIType>(BM->get<SPIRVExtInst>(Ops[BaseTypeIdx]));
+  size_t TotalCount = 1;
+  SmallVector<llvm::Metadata *, 8> Subscripts;
+  for (size_t I = SubrangesIdx; I < Ops.size(); ++I) {
+    auto *SR = transDebugInst<DISubrange>(BM->get<SPIRVExtInst>(Ops[I]));
+    if (auto *Count = SR->getCount().get<ConstantInt *>())
+      TotalCount *= Count->getZExtValue() > 0 ? Count->getZExtValue() : 0;
```

Comment:
`getZExtValue()` returns unsigned integer, so the ternary check is excessive here.

Reply:
should be `getSExtValue` IMHO

Reply:
Um, I thought we should either have
`TotalCount *= Count->getZExtValue();`
or
`TotalCount *= Count->getSExtValue() > 0 ? Count->getSExtValue() : 0;`

Did I understand the meaning wrong?

Reply:
Applied